lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -686,20 +855,20 @@ mod tests {
         let sequence_number = 1u8;
         let invalid_total_chunks = GELF_MAX_TOTAL_CHUNKS + 1;
         let payload = "foo";
-        let mut chunk = create_chunk(message_id, sequence_number, invalid_total_chunks, payload);
+        let mut chunk = create_chunk(message_id, sequence_number, invalid_total_chunks, &payload);
         let mut decoder = ChunkedGelfDecoder::default();
 
         let frame = decoder.decode_eof(&mut chunk);
         let error = frame.unwrap_err();
         let downcasted_error = downcast_framing_error(&error);
-        assert_eq!(
-            *downcasted_error,
+        assert!(matches!(
```

Comment:
changed  `assert_eq` to `assert!(matches!(..))` because `ChunkedGelfDecoderError` does not implement `Eq` and `PartialEq` now (due to `std::io::Error` limitations)